### DISCUSSION TITLE
MS Entra ID Documentation

### DISCUSSION THREAD

**Discussion by olanok** [ansibleguy76/ansibleforms#188] [Q&A] [answered]
Is there any documentation on configuring Ansible Forms with MS Entra ID?

**Comment by ansibleguy76** [comment]
### In a nutshell

**Register your app in azure (enterprise app)**

https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-register-app

- register the client id
- create a secret key

**Write down the client id & secret id**

- add api permissions

**Required API Permissions**

    Delegated User.Read
    Delegated GroupMember.Read.All

- add a group claim

**Required Group Claims**

    Security Groups
    Access > samAccountName

- add 1 or more valid call url's

Callback Url : https://_your_ansible_forms_url_/api/v1/auth/azureadoauth2/callback

**Comment by ansibleguy76** [comment, marked as answer]
### With screenshots

**Register a new app**

![image](https://github.com/ansibleguy76/ansibleforms/assets/89778474/385012a4-c49c-460e-ab5d-a98ec8e850c2)

**Add 1 or more redirect urls, to redirect back to ansible forms**

![image](https://github.com/ansibleguy76/ansibleforms/assets/89778474/fd793745-3676-4cf3-b5bc-0f860f1b5a4b)

**Write down the client ID !**

![image](https://github.com/ansibleguy76/ansibleforms/assets/89778474/f1321c44-3347-4eb3-a480-9df00d03ad94)

**Create a new secret and write it down !**

![image](https://github.com/ansibleguy76/ansibleforms/assets/89778474/d78b1b79-b91b-4901-98e7-96f6610d0d7d)

**Add api permissions**

![image](https://github.com/ansibleguy76/ansibleforms/assets/89778474/2c0b8ec7-a52e-4165-83af-a50e6e49a46c)

**Add group claims**

![image](https://github.com/ansibleguy76/ansibleforms/assets/89778474/be5800ac-2d5f-4b87-a3fb-ffdfbe5103d1)

**In ansibleforms fill in the public root url (examples `https://ansibleforms.domain.local` or `https://172.16.50.4:8443`)**

![image](https://github.com/ansibleguy76/ansibleforms/assets/89778474/adec8d0b-e5b0-4211-be74-b45699abf2df)

**Enter the client id and secret it and double check the callback url**

![image](https://github.com/ansibleguy76/ansibleforms/assets/89778474/0b5cc5b9-c0b2-4a44-aa00-2588f30b3321)

**in your AnsibleForms roles, add the groups and/or users you want, prefixed with `azuread/`**

![image](https://github.com/ansibleguy76/ansibleforms/assets/89778474/d655adbb-1c74-472d-b87b-79b5bf116643)
